Patches/Mechanoid Bench 3 JGH/Reverse_Patch_MB3.xml

```diff
@@ -0,0 +1,125 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<Patch>
+
+    <Operation Class="PatchOperationFindMod">
+        <mods>
+            <li>Mechanoid Bench 3</li>
+        </mods>
+        <match Class="PatchOperationSequence">
+            <operations>
+
+                <!-- Vanilla Mechanoid Weapon -->
+                <li Class="PatchOperationConditional">
+                    <xpath>/Defs/ThingDef[defName="Gun_InfernoCannon" or @Name="Gun_ChargeBlasterHeavyBase"]/destroyOnDrop</xpath>
+                    <nomatch Class="PatchOperationAdd">
+                        <xpath>/Defs/ThingDef[defName="Gun_InfernoCannon" or @Name="Gun_ChargeBlasterHeavyBase"]</xpath>
+                        <value>
+                            <destroyOnDrop>true</destroyOnDrop>
+                        </value>
+                    </nomatch>
+                </li>
+
+                <li Class="PatchOperationConditional">
+                    <xpath>/Defs/ThingDef[@Name="Gun_ChargeBlasterHeavyBase"]/tradeability</xpath>
+                    <nomatch Class="PatchOperationAdd">
+                        <xpath>/Defs/ThingDef[@Name="Gun_ChargeBlasterHeavyBase"]</xpath>
+                        <value>
+                            <tradeability>None</tradeability>
+                        </value>
+                    </nomatch>
+                    <match Class="PatchOperationReplace">
+                        <xpath>/Defs/ThingDef[@Name="Gun_ChargeBlasterHeavyBase"]/tradeability</xpath>
+                        <value>
+                            <tradeability>None</tradeability>
+                        </value>
+                    </match>
+                </li>
+
+                <li Class="PatchOperationConditional">

```

Comment:
I'm not sure precisely what's going wrong, but there's an issue in this operation here that's resulting in `Gun_InfernoCannon` having a duplicate `tradeability` tag.

Reply:
the main mod tried to make it tradeable through internal patches, I attempt to cancel it through another patch overwriting theirs, I might need to have a better approach for it

Reply:
I'm afraid any wandering trader especially combat one would be selling those, it's gonna be quite the fun if player ever interact with the weapon sold by them

Reply:
May require some loadorder requirements, perhaps.

Reply:
I'll try figuring it out, if it only affect Gun_InfernoCannon then only the part where it alters Gun_InfernoCannon might need a review/fix